### PR TITLE
Documented search_quote_analyzer in mapping types and detailed how to…

### PR DESCRIPTION
**Pull request by imranazad**
… disable stemming and stop words as a potential use case

### CODE REVIEW

**Comment by jpountz** [comment]
Documenting search_quote_analyzer is nice but the example is too hacky to be in the reference documentation in my opinion.

**Comment by imranazad** [comment]
@jpountz Thanks for the feedback. I'm assuming the disable_tokenization analyzer made you feel it's hacky? I guess it is so in that case I can remove that and just point it to a standard analyzer. With this approach you get the same result and it still clearly shows you the purpose of the search_quote_analyzer. Oh and also I forgot to include queries that illustrate how stems can be disabled for phrases so I can include those as well. Coming from a Microsoft FAST background and having experience in search, disabling stop words and stems for phrases is a big plus for Elastic in terms of firstly it's possible to do and secondly how easy it is and I think it's a real shame that the documentation currently lacks this information. The ability to disable stop words and stems can make all the difference to a user when reviewing which search product to adopt. If this information is clearly available in the documentation it can hold Elastic in good stead. Let me know what you think.

**Comment by jpountz** [comment]
@imranazad +1 on using an example with stop words to illustrate use-cases of the search_quote analyzer

**Comment by imranazad** [comment]
@jpountz Hey Adrian, sorry for the delay I caught a nasty cold so have been in bed for the past few days. I've made a few changes to the documentation, I've removed the hacky implementation and also included examples of disabling stems and stop words on phrases searched. Please let me know what you thinks. Thanks.

**Comment by jpountz** [comment]
No worries! Could we simplify a bit by removing the multi field and doing eg.

```
PUT /my_index
{
   "settings":{
      "analysis":{
         "analyzer":{
            "my_stop_analyzer":{
               "type":"custom",
               "tokenizer":"standard",
               "filter":[
                  "lowercase",
                  "english_stop"
               ]
            },
            "my_analyzer":{
               "type":"custom",
               "tokenizer":"standard",
               "filter":[
                  "lowercase"
               ]
            }
         },
         "filter":{
            "english_stop":{
               "type":"stop",
               "stopwords":"_english_"
            }
         }
      }
   },
   "mappings":{
      "my_type":{
         "properties":{
            "title":
               "type":"string",
               "analyzer":"my_analyzer",
               "analyzer":"my_stop_analyzer",
               "search_quote_analyzer":"my_analyzer"
            }
         }
      }
   }
}
```

If I'm not mistaken this way stop words will only be taken into account for phrases.

**Comment by imranazad** [comment]
@jpountz Hey Adrian, I've had a look at your suggestion and that wouldn't ever work with regard to a use case for stop words when dealing with phrases. You are correct stop words will only be taken into account for phrases however since the document is indexed with `my_stop_analyzer` the phrase query will never return any results because the original tokens have been filtered on stop words. So for example a search for `The quick brown fox` won't return any results as the `the` token isn't in the index.

**Comment by jpountz** [comment]
Woops sorry I had a mistake in my example, the second analyzer should read `search_analyzer` instead of `analyzer`. I just updated the example.

**Comment by imranazad** [comment]
@jpountz No worries. :-) Yes I see now, that would work, I love the simplicity and I remember now I did play around with this idea however that setup wouldn't be possible to replicate for both stop words and stemming would it using a single field? Previously I think I tried to look into this and gave up so resorted to `multi_fields`. Disabling stop words and stemming go together when it comes to phrase searches just like the way Google does it that's why I feel it's important that they be documented together. So for example if we had two documents with the titles `I bought two guard dogs` and `I bought one guard dog`, phrases search would work fine however a non-phrase query for `guard dogs` would never return the document titled  `I bought two guard dogs` since the `search_analyzer` would remove the stem and the original token would be indexed as `dogs`.

What do you think?

**Comment by jpountz** [comment]
> however that setup wouldn't be possible to replicate for both stop words and stemming would it using a single field?

True but I think examples in the reference documentation should be as simple as possible given that the goal is just to illustrate the feature. Additionally I don't really like the multi_field example as it also runs the phrase on the stemmed field, Ideally we would need a way to redirect phrases to a different field.

**Comment by imranazad** [comment]
@jpountz Ah I see, that's fine, not a problem, I'll make those changes you suggested then.

**Comment by imranazad** [comment]
@jpountz Hey Adrian, I finally got round to making the changes you suggested, all done now. Let me know what you think. Thanks.

File: docs/reference/mapping/params/search-quote-analyzer.asciidoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+               "filter":[
+                  "lowercase"
+               ]
+            },
+            "my_stop_analyzer":{ <2>
+               "type":"custom",
+               "tokenizer":"standard",
+               "filter":[
+                  "lowercase",
+                  "minimal_english"
```
Comment (jpountz) [inline docs/reference/mapping/params/search-quote-analyzer.asciidoc]: did you mean to add a stop filter rather than a stemmer?

**Comment by jpountz** [comment]
@imranazad thanks, this looks good. I think there is just a left-over in the search analyser definition which uses a stemmer instead of a stop filter?

**Comment by imranazad** [comment]
@jpountz Argh, my bad! Thanks, I've sorted it now. :-)

**Comment by jpountz** [comment]
:+1: I'll merge soon!

**Comment by imranazad** [comment]
@jpountz You're a star, many thanks Adrien, I must say I'm very excited about my first contribution to Elastic, thanks for the opportunity. :-)

**Comment by jpountz** [comment]
I merged the pr manually.

> thanks for the opportunity

You're welcome, thanks for helping!

**Comment by imranazad** [comment]
@jpountz Thanks pal.

**Pull request closed without merging**